### ISSUE TITLE
Static local functions can't see itself

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1435]
The below code doesn't compile:

```
		public static int Main(String[] args)
		{
			static void LocalStatic(int i = 0)
			{
				if (i < 10) LocalStatic(i + 1);
			}

			LocalStatic();
			return 0;
		}
```
Tested with https://github.com/beefytech/Beef/commit/c8ba4a9e4273c6fb7c56cbff0d6874cd1b48aa5f

**Comment by disarray2077** [comment]
It also can't see the newly added local static variables btw, I guess it would make sense if they could be seen by it since it wouldn't require a capture.

**Comment by bfiete** [comment]
Fixed at 9d2a22da4b3acfecb0220379f3ab241efe43cb88
Now you can also reference static/const local variables as well
